From 4f38a7774026e5b091874dfa15ddd55b715517fe Mon Sep 17 00:00:00 2001
From: saravananm <saravananm@ami.com>
Date: Tue, 3 Oct 2023 15:45:59 +0530
Subject: [PATCH] Fix-For-Asset_Tag

Description :
Get Asset tag is not showing the setted asset tag value.

Tested:
Build Successfully and Verified,its Working As Expected.

Before fix:
msgubuntu@Dell-server-3:~$ ipmitool -H 10.0.98.38 -U root -P 0penBmc -I lanplus -C 17 raw 0x2c 0x06 0xdc 0x00 0x10
 dc 14 2e 2e 2e 2e 2e 2e 2e 2e 2e 2e 2e 2e 2e 2e
 2e 2e

msgubuntu@Dell-server-3:~$ ipmitool -H 10.0.98.38 -U root -P 0penBmc -I lanplus -C 17 raw 0x2c 0x08 0xdc 0x00 0x10 0x31 0x32 0x33 0x34 0x35 0x36 0x37 0x38 0x39 0x30 0x31 0x32 0x33 0x34 0x35 0x36
 dc 10

msgubuntu@Dell-server-3:~$ ipmitool -H 10.0.98.38 -U root -P 0penBmc -I lanplus -C 17 raw 0x2c 0x06 0xdc 0x00 0x10
 dc 14 2e 2e 2e 2e 2e 2e 2e 2e 2e 2e 2e 2e 2e 2e
 2e 2e

After Fix :

msgubuntu@Dell-server-3:~$ ipmitool -H 10.0.98.38 -U root -P 0penBmc -I lanplus -C 17 raw 0x2c 0x06 0xdc 0x00 0x10
 dc 14 2e 2e 2e 2e 2e 2e 2e 2e 2e 2e 2e 2e 2e 2e
 2e 2e

msgubuntu@Dell-server-3:~$ ipmitool -H 10.0.98.38 -U root -P 0penBmc -I lanplus -C 17 raw 0x2c 0x08 0xdc 0x00 0x10 0x31 0>
 dc 10

msgubuntu@Dell-server-3:~$ ipmitool -H 10.0.98.38 -U root -P 0penBmc -I lanplus -C 17 raw 0x2c 0x06 0xdc 0x00 0x10
dc 10 31 32 33 34 35 36 37 38 39 30 31 32 33 34
 35 36

Working As Expected.

Signed-off-by: saravananm <saravananm@ami.com>
---
 dcmihandler.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dcmihandler.cpp b/dcmihandler.cpp
index 79bf850..a5d39b5 100644
--- a/dcmihandler.cpp
+++ b/dcmihandler.cpp
@@ -573,7 +573,7 @@ ipmi::RspType<uint8_t>   // Total Asset Tag Length
                             (assetTag.size() - offset),
                             assetTagDataStr.c_str(),
                             bytes);
-
+        dcmi::writeAssetTag(assetTag);
         tagLength = assetTag.size();
         return ipmi::responseSuccess(tagLength);
     }
-- 
2.25.1

